tools/dbTools.py

```diff
@@ -1,54 +1,177 @@
+        """Try to execute the SQL query passed by the query parameter, with the
+        arguments passed by the tuple argument params.
+        """
         try:
             self.cursor.execute(query, params)
-        except sqlite3.Error as error:
+            self.connection.commit()
+        except psycopg2.Error as e:
             self.logger.error(
                 'An exception had been encountered when executing %s',
                 query,
             )
             raise
 
     def _check_db(self):
+        """Create the tables needed by the web scraper if they don't exists."""
         self._execute(
             """
             CREATE TABLE IF NOT EXISTS article
             (
-                id INTEGER PRIMARY KEY,
-                title VARCHAR(64),
+                id SERIAL PRIMARY KEY,
+                title VARCHAR(255),
                 url VARCHAR(255),
                 file_hash VARCHAR(32),
-                scrap_again INTEGER(1) DEFAULT(0),
-                timestamp DATETIME DEFAULT CURRENT_TIMESTAMP
+                scrap_again BOOLEAN DEFAULT FALSE,
+                timestamp TIMESTAMP DEFAULT CURRENT_TIMESTAMP
             );
             """
         )
+        self._execute(
+            """
+            CREATE TABLE IF NOT EXISTS spiders
+            (
+                id SERIAL PRIMARY KEY,
+                name VARCHAR(64),
+                uuid VARCHAR(64),
+                status VARCHAR(255),
+                end_time TIMESTAMP,
+                start_time TIMESTAMP DEFAULT CURRENT_TIMESTAMP
+            );
+            """
+        )
+        self.connection.commit()
+
+    def reset_scraped(self):
+        """Set all the articles `scrap_again` attribute to 1, forcing the web
+        scraper to download and analyse them again.
+        """
+        self._execute(
+            "UPDATE article SET scrap_again = %s",
+            ('True',)
+        )
+
+    def clear_db(self):
+        """Remove all the articles from the database."""
+        self._execute(
+            "DELETE FROM article"
+        )
 
     def is_scraped(self, file_hash):
+        """Check if an article had already been scraped by looking for its file
+        hash into the database. Is the `scrap_again` attribute is true, return
+        False anyway.
+        """
         self._execute(
-            "SELECT id FROM article WHERE file_hash = ?",
+            """
+            SELECT id
+            FROM article
+            WHERE file_hash = %s AND scrap_again IS NOT TRUE;
+            """,
             (file_hash,)
         )
         result = self.cursor.fetchone()
         return result or None
 
+    def get_articles(self, offset=0, limit=-1):
+        """Return a list of articles. By default, returns every articles. This
+        method accepts start and end arguments to paginate results.
+        """
+        if limit > 0:
+            self._execute(
+                "SELECT title, file_hash, url FROM article LIMIT %s OFFSET %s",
```

Comment:
also maybe an overkill for this version but worth moving to an object-relational mapper in the future with a model to remove sql and improve readability and robustness.

Reply:
Agreed. It started quite small, but it's growing and would definitely benefits from it now.